src/Recording.ts

```diff
@@ -172,6 +173,11 @@ export default class Recording {
   }
 
   private emit(event: unknown) {
+    // Check here if we should record instead of requiring each
+    // possible hook to check it.
+    // Checking this inside hooks too, will save CPU cycles only
+    // by preventing unnecessary event creation.
```

Comment:
I think it was already fine as it was — `record()` method already checked if `shouldRecord()`; I just thought it would be good to say that in this comment :)

Reply:
Just to clarify, all the hooks which call event creation methods directly (sqlite, prisma, pg, mysql, mongo, http) will create the event objects, but the events will not be emitted when we should not record. I believe you meant these by some wasted events.

Reply:
perf commit rolled back.